### ISSUE TITLE
window.postMessage is undefined

### ISSUE THREAD

**Issue by RameshRM** [tmpvar/jsdom#1006]
Im using jsdom for iFrame postMessage testing, postMessage is "undefined" , any idea how can i go about fixing this ?

**Comment by domenic** [comment]
Sure! A pull request to implement cross-document messaging functionality in jsdom would be awesome. You should be able to find the definitions for HTMLFrameElement and HTMLIFrameElement in the source. [Contributing.md](https://github.com/tmpvar/jsdom/blob/master/Contributing.md) has more details on where to find tests and such. Let me know if you need anything else!

**Comment by alvarorahul** [comment]
hi @RameshRM, are you working on getting this feature in? It would be super useful for some of the stuff I'm working on :)

**Comment by jeffcarp** [comment]
@RameshRM @alvarorahul I don't want to step on any toes, but if you don't have the time, I'd be interested in picking this up.

**Comment by chrisbarmonde** [comment]
I guess we've all decided to start our `postMessage` testing on the same day and found the same problem! I'm not smart enough to implement this, but I eagerly anticipate not having to mock all this crap away in my tests.

**Comment by dariocravero** [comment]
Hi,

I don't know if there are any news on this but we got a very basic implementation working which just wraps the message as the `data` in the event and emits the event on the `window` it was bound to.
It's not enforcing `targetOrigin` nor it's implementing `transfer` but perhaps it helps as a starting point?

```js
function postMessage(message, targetOrigin, transfer) {
  let event = this.document.createEvent('messageevent');
  event._type = 'message';
  event.data = message;
  this.dispatchEvent(event);
}

// Use it
window.postMessage = postMessage.bind(window);
```

Cheers,
Darío

**Comment by domenic** [comment]
This is in progress as https://github.com/tmpvar/jsdom/pull/1140 actually.

**Comment by dariocravero** [comment]
Amazing! :) Thanks @domenic

**Comment by domenic** [comment]
Merged in #1140! A release should be coming shortly.

**Comment by chrisbarmonde** [comment]
Thanks, @domenic!
